### PR TITLE
changed boosted performance

### PR DESCRIPTION
**Pull request by mbrueschke**
@plentymarkets/team-multichannel-koboldmaki

### CODE REVIEW

File: src/ResultField/KuponaDE.php [hunk 1]
```diff
@@ -151,4 +190,105 @@ public function generateResultFields(array $formatSettings = []):array
...
+					'urlPreview',
+					'urlSecondPreview',
+					'url',
+					'path',
+					'position',
+				],
+
+				// texts
+				'texts' => [
+					'urlPath',
```
Comment (alinmigea) [inline src/ResultField/KuponaDE.php:286]: maybe you could add the 'lang' field also. usually is updated, but just to be sure

File: meta/documents/user_guide_de.md [hunk 2]
```diff
@@ -3,16 +3,281 @@
...
+		</td>
+		<td>
+			<b>Inhalt:</b> Die <b>Grundpreisinformation</b> im Format "Preis / Einheit". (Beispiel: 10,00 EUR / Kilogramm)
+		</td>        
+	</tr>
+</table>
+
+## 4 Lizenz
+
+Das gesamte Projekt unterliegt der GNU AFFERO GENERAL PUBLIC LICENSE – weitere Informationen finden Sie in der [LICENSE.md](https://github.com/plentymarkets/plugin-elastic-export-rakuten-de/blob/master/LICENSE.md).
```
Comment (alinmigea) [inline meta/documents/user_guide_de.md]: change the link to kupona-de

File: meta/documents/user_guide_en.md [hunk 3]
```diff
@@ -3,16 +3,280 @@
...
+		<td>
+			<b>Content:</b> The <b>base price information</b>. The format is "price / unit". (Example: 10,00 EUR / kilogram)
+		</td>        
+	</tr>
+</table>
 
 ## License
 
-This project is licensed under the GNU AFFERO GENERAL PUBLIC LICENSE.- find further information in the [LICENSE.md](https://github.com/plentymarkets/plugin-elastic-export-kupona-de/blob/master/LICENSE.md).
+This project is licensed under the GNU AFFERO GENERAL PUBLIC LICENSE.- find further information in the [LICENSE.md](https://github.com/plentymarkets/plugin-elastic-export-rakuten-de/blob/master/LICENSE.md).
```
Comment (alinmigea) [inline meta/documents/user_guide_en.md]: here also :)

File: meta/documents/changelog_de.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # Release Notes für Elastic Export Kupona.de
 
+## v1.0.5 (2017-08-11)
```
Comment (alinmigea) [inline meta/documents/changelog_de.md]: also maybe the dates should be changed, before releasing

File: src/Generator/KuponaDE.php [hunk 5]
```diff
@@ -45,112 +57,185 @@ public function __construct(ArrayHelper $arrayHelper)
...
+	/**
+	 * Builds a row for each variation
+	 *
+	 * @param $variation
+	 * @param $settings
+	 */
+	private function buildRow($variation, $settings)
+	{
+		// Get and set the price and rrp
+		$priceList = $this->elasticExportPriceHelper->getPriceList($variation, $settings);
```
Comment (alinmigea) [inline src/Generator/KuponaDE.php:170]: as a suggestion, shouldn't it be better to separate the logic and maybe move the price and currency in a function?

File: src/Generator/KuponaDE.php [hunk 6]
```diff
@@ -45,112 +57,185 @@ public function __construct(ArrayHelper $arrayHelper)
...
+		}
+
+		$currency = '';
+		if($price > 0)
+		{
+			$currency = $priceList['currency'];
+		}
+
+		// Get shipping costs
+		$shippingCost = $this->elasticExportHelper->getShippingCost($variation['data']['item']['id'], $settings);
```
Comment (alinmigea) [inline src/Generator/KuponaDE.php]: and maybe here for the shipping costs we could build a cache

File: src/Generator/KuponaDE.php [hunk 7]
```diff
@@ -45,112 +57,185 @@ public function __construct(ArrayHelper $arrayHelper)
...
+			$shippingCost = number_format((float)$shippingCost, 2, '.', '');
+		}
+		else
+		{
+			$shippingCost = '';
+		}
+
+		$imageList = $this->elasticExportHelper->getImageListInOrder($variation, $settings, 0, $this->elasticExportHelper::VARIATION_IMAGES, $this->elasticExportHelper::SIZE_NORMAL, true);
+
+		$previewUls = '';
```
Comment (alinmigea) [inline src/Generator/KuponaDE.php]: maybe here you could change it to previewUrls, you've eat the 'r'

Comment (alinmigea) [reply src/Generator/KuponaDE.php]: and maybe the whole logic in a function, if you still have the patience :)

File: src/Generator/KuponaDE.php [hunk 8]
```diff
@@ -45,112 +57,185 @@ public function __construct(ArrayHelper $arrayHelper)
...
+				$middleUrls = $middleUrls . ';'.$image['urlMiddle'];
+				$normalUrls = $normalUrls . ';'.$image['url'];
+			}
+
+			$iteration ++;
+		}
+
+		$data = [
+			'prod_number'           => $variation['id'],
+			'prod_name'             => $this->elasticExportHelper->getMutatedUrl($variation, $settings),
```
Comment (alinmigea) [inline src/Generator/KuponaDE.php]: here should it be getMutatedName?

File: src/ResultField/KuponaDE.php [hunk 9]
```diff
@@ -151,4 +190,105 @@ public function generateResultFields(array $formatSettings = []):array
...
+	}
+
+	/**
+	 * @return array
+	 */
+	private function getNestedKeyList()
+	{
+		return [
+			'keys' => [
+				// Attributes
```
Comment (alinmigea) [inline src/ResultField/KuponaDE.php:221]: here you've forgotten to add the 'texts' as keys

**Review by alinmigea — CHANGES_REQUESTED** [review]

File: src/ResultField/KuponaDE.php (hunk 1 above)
Comment (mbrueschke) [reply src/ResultField/KuponaDE.php:286]: no, because it wouldnt make any sense

File: src/ResultField/KuponaDE.php (hunk 9 above)
Comment (alinmigea) [reply src/ResultField/KuponaDE.php:221]: 👍

File: src/ResultField/KuponaDE.php (hunk 1 above)
Comment (alinmigea) [reply src/ResultField/KuponaDE.php:286]: good reminder :)

File: src/Generator/KuponaDE.php (hunk 5 above)
Comment (alinmigea) [reply src/Generator/KuponaDE.php:170]: as discussed, the logic for price and currency remains here.

File: meta/documents/changelog_de.md (hunk 4 above)
Comment (alinmigea) [reply meta/documents/changelog_de.md]: @mbrueschke please change the release date for tomorrow

**Review by alinmigea — APPROVED** [review]

**Pull request merged**